packages/cozy-konnector-libs/libs/linkBankOperations.spec.js

```diff
@@ -285,4 +285,110 @@ describe('linker', () => {
+        { _id: 'b3', originalDate: '2018-03-10T00:00:00Z' },
+        { _id: 'b4', originalDate: '2018-03-15T00:00:00Z' },
+        { _id: 'b5', originalDate: '2018-03-15T00:00:00Z' },
+        { _id: 'b6', originalDate: '2018-03-20T00:00:00Z' },
+        { _id: 'b7', originalDate: '2018-03-20T00:00:00Z' },
+        { _id: 'b8', originalDate: '2018-03-20T00:00:00Z' },
+        { _id: 'b9', originalDate: '2018-03-20T00:00:00Z' },
+        { _id: 'b10', originalDate: '2018-03-30T00:00:00Z' }
+      ]
+
+      test('it groups bills by their originalDate property', () => {
+        const result = linker.groupBillsByOriginalDate(bills)
+
+        expect(result['2018-03-10T00:00:00Z']).toHaveLength(3)
+        expect(result['2018-03-15T00:00:00Z']).toHaveLength(2)
+        expect(result['2018-03-20T00:00:00Z']).toHaveLength(4)
+        expect(result['2018-03-30T00:00:00Z']).toHaveLength(1)
+      })
+    })
+
+    describe('generateBillsCombinations', () => {
+      const bills = [
+        { _id: 'b1' },
+        { _id: 'b2' },
+        { _id: 'b3' },
+        { _id: 'b4' }
+      ]
+
+      test('it generates the right combinations', () => {
+        const result = linker.generateBillsCombinations(bills)
+
+        expect(result).toContainEqual([bills[0], bills[1]])
+        expect(result).toContainEqual([bills[0], bills[2]])
+        expect(result).toContainEqual([bills[0], bills[3]])
+        expect(result).toContainEqual([bills[1], bills[2]])
+        expect(result).toContainEqual([bills[1], bills[3]])
+        expect(result).toContainEqual([bills[2], bills[3]])
+        expect(result).toContainEqual([bills[0], bills[1], bills[2]])
+        expect(result).toContainEqual([bills[0], bills[1], bills[3]])
+        expect(result).toContainEqual([bills[0], bills[2], bills[3]])
+        expect(result).toContainEqual([bills[1], bills[2], bills[3]])
+        expect(result).toContainEqual([bills[0], bills[1], bills[2], bills[3]])
+      })
+    })
+
+    describe('combineBills', () => {
+      const bills = [
+        {
+          _id: 'b1',
+          amount: 10,
+          originalAmount: 20,
+          originalDate: '2018-03-10T00:00:00Z'
+        },
+        {
+          _id: 'b2',
+          amount: 10,
+          originalAmount: 10,
+          originalDate: '2018-03-10T00:00:00Z'
+        }
+      ]
+
+      test('it generate a bill with the right amount', () => {
+        const combinedBill = linker.combineBills(...bills)
+        expect(combinedBill.amount).toBe(20)
+      })
+
+      test('it generates a bill with the right originalAmount', () => {
+        const combinedBill = linker.combineBills(...bills)
+        expect(combinedBill.originalAmount).toBe(30)
+      })
+
+      test('it generates a bill with the right originalDate', () => {
+        const combinedBill = linker.combineBills(...bills)
+        expect(combinedBill.originalDate).toBe('2018-03-10T00:00:00Z')
+      })
+    })
+  })
```

Comment:
There are no tests that tries to link with groups (more like integration tests). I think it should be good to add them. They will be more robust that the unit tests.